File: Datastructures/hash_table_design/hash_table.py
from collections import deque

class MyHashSet:

    def __init__(self):
        self.num_buckets = 1029
        self.buckets = [deque() for _ in range(self.num_buckets)]
    
    def getBucket(self,key) -> int:
        return key%self.num_buckets
          

    def add(self, key: int) -> None:
        if  self.contains(key):
            return
        self.buckets[self.getBucket(key)].append(key)
            
        
    def remove(self, key: int) -> None:
        if self.contains(key):
            self.buckets[self.getBucket(key)].remove(key)
        

    def contains(self, key: int) -> bool:
        for i in self.buckets[self.getBucket(key)]:
            if key == i:
                return True 
            
        return False

File: Datastructures/hash_table_design/hash_table.js

class MyHashSet {
    #num_buckets = 1029;
    #buckets = Array.from({length: this.#num_buckets}, () => []);

    getBucket(key) {
        return key % this.#num_buckets;
    }

    add(key) {
        if (this.contains(key)) {
            return;
        }
        this.#buckets[this.getBucket(key)].push(key);
    }

    remove(key) {
        let bucket = this.#buckets[this.getBucket(key)];
        let index = bucket.indexOf(key);
        if (index !== -1) {
            bucket.splice(index, 1);
        }
    }

    contains(key) {
        let bucket = this.#buckets[this.getBucket(key)];
        return bucket.includes(key);
    }
}
